### PR TITLE
Replace use of gleam in webgl with sparkle.

### PR DESCRIPTION
**Pull request by jdm**
Rely on https://github.com/servo/sparkle for the GL bindings required to implement Servo's WebGL stack. This does not touch the GL usage in the compositor or webrender, which continue to rely on https://github.com/servo/gleam. This means that any breaking changes to the public GL bindings interface (such as required by https://github.com/servo/servo/pull/24111) only require updating rust-offscreen-rendering-context and rust-webvr, rather than many other crates that are out of our direct control.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24176)

### CODE REVIEW

File: components/servo/lib.rs [hunk 1]
```diff
@@ -412,7 +412,11 @@ where
         } else {
             let dispatcher =
                 Box::new(MainThreadDispatcher::new(compositor_proxy.clone())) as Box<_>;
-            GLContextFactory::current_native_handle(dispatcher, window.gl().get_type())
+            let gl_type = match window.gl().get_type() {
+                gl::GlType::Gl => sparkle::gl::GlType::Gl,
+                gl::GlType::Gles => sparkle::gl::GlType::Gles,
+            };
```
Comment (asajeffrey) [inline components/servo/lib.rs]: Can we re-export GlType to avoid this type conversion?

**Review by asajeffrey — COMMENTED** [review]
Looks good so far! You can r=me once the dependencies have merged.

File: components/servo/lib.rs (hunk 1 above)
Comment (jdm) [reply components/servo/lib.rs]: Re-export where? Have gleam depend on sparkle? Have sparkle depend on gleam? Neither solution sounds like an improvement to me.

Comment (asajeffrey) [reply components/servo/lib.rs]: Ah, we're getting the GlType from glutin, oh that makes sense.

**Comment by jdm** [comment]
@bors-servo retry
* wpt run timed out on mac3

**Comment by kvark** [comment]
Trying to understand the reasoning here. Is sparkle essentially a fork of gleam that will have more breaking changes? why couldn't you just manage it as a fork of gleam then?

**Comment by jdm** [comment]
Sparkle doesn't use traits, so changes to expose new methods don't cause semver major API changes. Additionally, sparkle only exposes methods for APIs thare are actually available for GL and GLES.

**Pull request merged**